src/node.cpp

```diff
@@ -85,19 +85,27 @@ void publish_scan(ros::Publisher *pub,
     if (!inverted) { // assumes scan window at the top
         for (size_t i = 0; i < node_count; i++) {
             float read_value = (float) nodes[i].distance_q2/4.0f/1000;
+            float range = read_value * distance_factor;
+            if (range < scan_msg.range_min)
+                range = std::numeric_limits<float>::infinity();
```

Comment:
as long as we don't mistakenly clear obstacles on the costmap, that's good enough for me